mlflow/store/sqlalchemy_store.py

```diff
@@ -251,55 +259,86 @@ def create_experiment(self, name, artifact_location=None):
+        """
         stages = LifecycleStage.view_type_to_stages(view_type)
         conditions = [SqlExperiment.lifecycle_stage.in_(stages)]
-
         if ids and len(ids) > 0:
             int_ids = [int(eid) for eid in ids]
             conditions.append(SqlExperiment.experiment_id.in_(int_ids))
-
         if names and len(names) > 0:
             conditions.append(SqlExperiment.name.in_(names))
-        return session.query(SqlExperiment).filter(*conditions)
+
+        query_options = self._get_eager_experiment_query_options() if eager else []
+        return session \
+            .query(SqlExperiment) \
+            .options(*query_options) \
+            .filter(*conditions) \
+            .all()
 
     def list_experiments(self, view_type=ViewType.ACTIVE_ONLY):
         with self.ManagedSessionMaker() as session:
             return [exp.to_mlflow_entity() for exp in
-                    self._list_experiments(session=session, view_type=view_type)]
+                    self._list_experiments(session=session, view_type=view_type, eager=True)]
 
-    def _get_experiment(self, session, experiment_id, view_type):
+    def _get_experiment(self, session, experiment_id, view_type, eager=False):
+        """
+        :param eager: If ``True``, eagerly loads the experiments's tags. If ``False``, these tags
+                      are not eagerly loaded and will be loaded if/when their corresponding
+                      object properties are accessed from the resulting ``SqlExperiment`` object.
+        """
         experiment_id = experiment_id or SqlAlchemyStore.DEFAULT_EXPERIMENT_ID
-        experiments = self._list_experiments(
-            session=session, ids=[experiment_id], view_type=view_type).all()
-        if len(experiments) == 0:
+        stages = LifecycleStage.view_type_to_stages(view_type)
+        query_options = self._get_eager_experiment_query_options() if eager else []
+
+        experiment = session \
+            .query(SqlExperiment) \
+            .options(*query_options) \
+            .filter(
+                SqlExperiment.experiment_id == experiment_id,
+                SqlExperiment.lifecycle_stage.in_(stages)) \
+            .one_or_none()
+
+        if experiment is None:
             raise MlflowException('No Experiment with id={} exists'.format(experiment_id),
                                   RESOURCE_DOES_NOT_EXIST)
-        if len(experiments) > 1:
-            raise MlflowException('Expected only 1 experiment with id={}. Found {}.'.format(
-                experiment_id, len(experiments)), INVALID_STATE)
 
-        return experiments[0]
+        return experiment
+
+    @staticmethod
+    def _get_eager_experiment_query_options():
+        """
+        :return: A list of SQLAlchemy query options that can be used to eagerly load the following
+                 experiment attributes when fetching an experiment: ``tags``.
+        """
+        return [
+            sqlalchemy.orm.joinedload(SqlExperiment.tags),
```

Comment:
Do you need to add `lazy` argument while setting up this `relationship` in db entities for `SqlExperiment` and `SqlTag` for this `joinedload` to work?

https://docs.sqlalchemy.org/en/13/orm/loading_relationships.html

Also `joinedload` takes type of join as argument. We prolly need a `LEFT OUTER`?? Such that if tags are missing we still get experiments? Or not?

https://engineering.shopspring.com/speed-up-with-eager-loading-in-sqlalchemy-87a176cfd7ad

Also is there a way to test that indeed database is queried eagerly and store does not go back to database for tags, in this case

Reply:
Great questions!

1. The `lazy` argument does not need to be specified while setting up the backref relationship in the `SqlExperimentTag` model in order to specify a different loading procedure at query execution time. By default (e.g., when `lazy` is not specified), backref relationships are configured for lazy loading. However, by specifying the `joinedload` option at query time, related data can be resolved eagerly instead. The reason that I opted not to configure the relationship for joined loads within the DB entity definitions is that most experiment operations (e.g., SetExperimentTag, DeleteExperiment, and any operation that fetches an experiment to check its activity state - such as StartRun) do not require experiment tag data to be loaded.

2. `sqlalchemy.orm.joinedload` does not take a join type as an argument. Rather, the function defines an `innerjoin` parameter that is `False` by default (see https://docs.sqlalchemy.org/en/13/orm/loading_relationships.html#sqlalchemy.orm.joinedload). If `innerjoin` is `False`, then a `LEFT OUTER` join is used. Therefore, the current invocation of `sqlalchemy.orm.joinedload` produces a left outer join, as desired.

3. Great question! The SQLAlchemy documentation recommends the use of Python's `cProfile` library to profile SQLAlchemy operations and analyze query paths (see https://docs.sqlalchemy.org/en/13/faq/performance.html#how-can-i-profile-a-sqlalchemy-powered-application). 

Accordingly, I defined the following Python script and saved it in a file called `profile.py`:
```
import cProfile
import StringIO
import pstats
import contextlib

@contextlib.contextmanager
def profiled():
    pr = cProfile.Profile()
    pr.enable()
    yield
    pr.disable()
    s = StringIO.StringIO()
    ps = pstats.Stats(pr, stream=s).sort_stats('cumulative')
    ps.print_stats()
    # uncomment this to see who's calling what
    # ps.print_callers()
    print(s.getvalue())


import mlflow
from mlflow.tracking.client import MlflowClient

mlflow.set_tracking_uri("sqlite:///mytestdb")

client = MlflowClient()

for i in range(50):
    client.create_experiment("experiment_{}".format(i))

with profiled():
    client.list_experiments()
```

I then ran the following command on the current `master` branch (without these changes):
```
$ python profile.py > old_profile.txt
```
and I ran the following command on the PR branch (with the eager loading optimizations):
```
$ python profile.py > new_profile.txt
```

Here are the contents of `old_profile.txt`:
```
         15438 function calls (15291 primitive calls) in 0.025 seconds

   Ordered by: cumulative time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.000    0.000    0.025    0.025 /Users/czumar/mlflow/mlflow/tracking/client.py:96(list_experiments)
        1    0.000    0.000    0.025    0.025 /Users/czumar/mlflow/mlflow/store/sqlalchemy_store.py:266(list_experiments)
       51    0.000    0.000    0.021    0.000 /Users/czumar/mlflow/mlflow/store/dbmodels/models.py:68(to_mlflow_entity)
      256    0.000    0.000    0.020    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py:268(__get__)
   102/51    0.000    0.000    0.020    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py:658(get)
       51    0.000    0.000    0.019    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:655(_load_for_state)
       51    0.000    0.000    0.018    0.000 <string>:1(<lambda>)
       51    0.001    0.000    0.018    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:744(_emit_lazyload)
       51    0.000    0.000    0.015    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:525(all)
       51    0.001    0.000    0.010    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:410(__iter__)
       52    0.000    0.000    0.008    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3336(_execute_and_instances)
       52    0.000    0.000    0.005    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:922(execute)
      103    0.001    0.000    0.005    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/loading.py:35(instances)
       52    0.000    0.000    0.005    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:285(_execute_on_connection)
       52    0.000    0.000    0.005    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1056(_execute_clauseelement)
       52    0.000    0.000    0.004    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1163(_execute_context)
       52    0.000    0.000    0.003    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4151(row_processor)
       52    0.002    0.000    0.003    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/loading.py:337(_instance_processor)
       52    0.000    0.000    0.002    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3351(_get_bind_args)
       51    0.000    0.000    0.002    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:618(_generate_lazy_clause)
       52    0.000    0.000    0.002    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3330(_connection_from_session)
        1    0.000    0.000    0.002    0.002 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3311(__iter__)
       51    0.000    0.000    0.002    0.000 /Users/czumar/anaconda2/lib/python2.7/copy.py:66(copy)
       51    0.000    0.000    0.002    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:2772(_get_state_attr_by_column)
       52    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1047(connection)
       51    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/annotation.py:102(__eq__)
       52    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:551(do_execute)
       53    0.001    0.000    0.001    0.000 {method 'execute' of 'sqlite3.Cursor' objects}
       52    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:726(operate)
       52    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1126(_connection_for_bind)
   102/51    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/operators.py:358(__eq__)
       52    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:406(_connection_for_bind)
       52    0.001    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:638(_init_compiled)
       51    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py:1303(set_committed_value)
      102    0.000    0.000    0.001    0.000 {operator.eq}
       51    0.001    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/copy.py:306(_reconstruct)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:2223(_contextual_connect)
       52    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:64(operate)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:2259(_wrap_pool_connect)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:345(connect)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:748(_checkout)
        2    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:399(compile)
        2    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:464(_compiler)
        2    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:500(__init__)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:481(checkout)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/impl.py:236(_do_get)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:296(_create_connection)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:425(__init__)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:622(__connect)
        2    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:274(__init__)
        2    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:349(process)
     20/2    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:85(_compiler_dispatch)
        2    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1999(visit_select)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/strategies.py:106(connect)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:452(connect)
        1    0.001    0.001    0.001    0.001 {_sqlite3.connect}
       52    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/default_comparator.py:41(_boolean_compare)
      158    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/base.py:215(generate)
       51    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/loading.py:492(_instance)
       51    0.001    0.000    0.001    0.000 {method '__reduce_ex__' of 'object' objects}
       52    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3066(__init__)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1761(_label_select_column)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:183(execution_options)
        2    0.000    0.000    0.000    0.000 /Users/czumar/mlflow/mlflow/store/sqlalchemy_store.py:145(make_managed_session)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py:1183(_initialize_collection)
  140/134    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:852(__get__)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:1092(get_result_proxy)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3820(_compile_context)
       51    0.000    0.000    0.000    0.000 <string>:1(params)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/contextlib.py:21(__exit__)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:1195(fetchall)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:224(_bake)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:714(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:992(commit)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:490(commit)
       53    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/interfaces.py:556(create_row_processor)
      102    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:890(get)
      158    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:518(_clone)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:569(close)
      104    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:151(union)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/instrumentation.py:315(initialize_collection)
      719    0.000    0.000    0.000    0.000 {isinstance}
        1    0.000    0.000    0.000    0.000 /Users/czumar/mlflow/mlflow/store/sqlalchemy_store.py:254(_list_experiments)
      157    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:724(_getter)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:863(<lambda>)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:780(visit_label)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:740(_init_metadata)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:399(_bind_mapper)
       63    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/interfaces.py:516(_get_context_loader)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/state.py:771(_commit)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:1034(create_cursor)
      102    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:101(add_criteria)
       51    0.000    0.000    0.000    0.000 <string>:1(_set_lazyload_from)
       52    0.000    0.000    0.000    0.000 <string>:1(with_session)
    18/14    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:925(__getattr__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4186(setup_context)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1573(_autoflush)
       51    0.000    0.000    0.000    0.000 /Users/czumar/mlflow/mlflow/entities/experiment.py:13(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/loading.py:287(_setup_entity_query)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:869(close)
        5    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/deprecations.py:117(warned)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:987(close)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:831(_checkin)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/instrumentation.py:337(new_instance)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:51(__call__)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3086(__bool__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:2186(_compose_select_body)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:645(_finalize_fairy)
      319    0.000    0.000    0.000    0.000 {method 'copy' of 'dict' objects}
       53    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:1157(_next)
        9    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:440(dialect_impl)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:1708(params)
      102    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:380(with_post_criteria)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:507(checkin)
       55    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:2417(flush)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3945(_simple_statement)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:366(_return_conn)
      157    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:643(_getter)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/impl.py:233(_do_return_conn)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:529(close)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:615(__close)
        6    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:505(_dialect_info)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:257(_close_connection)
     1304    0.000    0.000    0.000    0.000 {setattr}
      4/1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/operators.py:515(in_)
      6/2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:97(_compiler_dispatch)
      114    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/inspection.py:38(inspect)
       10    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/interfaces.py:546(setup)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py:210(operate)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1539(query)
      3/1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/operators.py:1252(in_op)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:141(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/properties.py:314(operate)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:507(do_close)
      181    0.000    0.000    0.000    0.000 {getattr}
      104    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:681(self_group)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:170(_set_entities)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:46(_state_session)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1261(visit_binary)
        1    0.000    0.000    0.000    0.000 {method 'close' of 'sqlite3.Connection' objects}
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:60(__init__)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/state.py:379(_cleanup)
        2    0.000    0.000    0.000    0.000 <string>:1(select)
      153    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:85(_update_cache_key)
       56    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:755(unique_list)
        9    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:834(visit_column)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1416(get_bind)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1318(_generate_generic_binary)
       56    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:2463(_is_clean)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:1161(_fetchall_impl)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/loading.py:677(_populate_full)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:169(_clone)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:869(_soft_close)
       27    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:3529(quote)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/default_comparator.py:158(_in_impl)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:815(has_intersection)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/state.py:412(dict)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:2157(_setup_select_stack)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:130(__call__)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:641(construct_params)
      424    0.000    0.000    0.000    0.000 {built-in method __new__ of type object at 0x100a83bb8}
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/collections.py:618(__init__)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:883(begin)
      102    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:375(_using_post_criteria)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:949(cursor)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:3641(_columns_plus_names)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:2687(__init__)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:220(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:215(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:591(_memoized_attr__simple_lazy_clause)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:140(__new__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:300(_as_query)
       52    0.000    0.000    0.000    0.000 {method 'fetchall' of 'sqlite3.Cursor' objects}
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:524(adapt)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/contextlib.py:15(__enter__)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:121(for_session)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:71(get)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:3188(__call__)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:1136(constructor_copy)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:2988(_get_display_froms)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:302(__missing__)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/default_comparator.py:354(_check_literal)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:1178(process_rows)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:3646(name_for_col)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:205(__init__)
        1    0.000    0.000    0.000    0.000 <string>:1(__init__)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:947(should_autocommit)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:336(_take_snapshot)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:3477(_requires_quotes)
      171    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/attr.py:267(__bool__)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/state.py:89(__init__)
      423    0.000    0.000    0.000    0.000 {method 'append' of 'list' objects}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:2959(_froms)
       72    0.000    0.000    0.000    0.000 {hasattr}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:645(__init__)
       54    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:267(_assert_active)
      169    0.000    0.000    0.000    0.000 {method 'update' of 'dict' objects}
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:1430(to_instance)
       53    0.000    0.000    0.000    0.000 {method 'cursor' of 'sqlite3.Connection' objects}
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:3598(format_label)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:856(set_default_params)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:292(cloned_traverse)
      102    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:886(_inc_counter)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:771(<lambda>)
       56    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/identity.py:56(check_modified)
       53    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/operators.py:1399(is_boolean)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:384(_remove_snapshot)
      3/1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:299(clone)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:379(_entity_zero)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3972(_label)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:202(_set_entity_selectables)
       15    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/registry.py:53(_collection_gced)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/loading.py:880(for_context)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:334(_merge_cursor_description)
       50    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/identity.py:213(_fast_discard)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:717(visit_grouping)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3980(_gen_label)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1445(visit_bindparam)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/instrumentation.py:328(get_impl)
      102    0.000    0.000    0.000    0.000 {hash}
        5    0.000    0.000    0.000    0.000 {method 'join' of 'str' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:961(visit_clauselist)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/copy_reg.py:92(__newobj__)
       53    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:883(create_row_processor)
        9    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1542(_truncated_identifier)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:80(__getitem__)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:356(__init__)
       61    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/attr.py:255(__call__)
       54    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4131(_get_entity_clauses)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4037(__new__)
      329    0.000    0.000    0.000    0.000 {len}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:968(<genexpr>)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:970(<genexpr>)
      8/3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:273(get_cls_kwargs)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1278(close)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:1051(get_result_processor)
        6    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/base.py:38(_from_objects)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1327(_safe_close_cursor)
      224    0.000    0.000    0.000    0.000 {method 'get' of 'dict' objects}
       51    0.000    0.000    0.000    0.000 {sorted}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1323(_close_impl)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3102(_copy_internals)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:295(__get__)
        2    0.000    0.000    0.000    0.000 <string>:1(filter)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:802(to_list)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4071(setup_entity)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4019(_bind_param)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/copy_reg.py:95(_slotnames)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:326(_unbake_subquery_loaders)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:276(traverse)
       52    0.000    0.000    0.000    0.000 {zip}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1529(_truncate_bindparam)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:160(_effective_key)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:911(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:514(do_ping)
       55    0.000    0.000    0.000    0.000 {method 'acquire' of 'thread.lock' objects}
       52    0.000    0.000    0.000    0.000 {sqlalchemy.cutils._distill_params}
       62    0.000    0.000    0.000    0.000 {method 'intersection' of 'set' objects}
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:483(_cached_result_processor)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/annotation.py:78(_compiler_dispatch)
       16    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:56(remove)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:361(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:69(__init__)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4184(__new__)
       53    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/operators.py:1376(is_comparison)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:521(_gen_dialect_impl)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:119(_for_instance)
       75    0.000    0.000    0.000    0.000 {method 'values' of 'dict' objects}
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:864(_cursor_description)
       15    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:94(__getattr__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:1864(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:740(_memoized_attr__bakery)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:1002(_use_server_side_cursor)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:913(__setitem__)
       53    0.000    0.000    0.000    0.000 {method 'intersection' of 'frozenset' objects}
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:477(set_connection_execution_options)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:198(_clone)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:116(_for_class)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:222(iterate)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:1202(with_session)
      103    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/annotation.py:99(__hash__)
       55    0.000    0.000    0.000    0.000 {method 'release' of 'thread.lock' objects}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/base.py:346(_is_mapped_class)
       25    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:362(__getitem__)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:409(type_descriptor)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4308(_as_truncated)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/identity.py:158(_add_unpresent)
      163    0.000    0.000    0.000    0.000 {id}
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:352(__str__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4299(apply_map)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4642(__init__)
       53    0.000    0.000    0.000    0.000 {next}
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/identity.py:163(get)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/state.py:328(_detach_states)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/declarative/clsregistry.py:106(_remove_item)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/logging/__init__.py:1152(debug)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:371(_query_entity_zero)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:1729(filter)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:852(<lambda>)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:620(_bind_processors)
      102    0.000    0.000    0.000    0.000 {method 'extend' of 'list' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1711(commit)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:1440(adapt_type)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:935(oneshot)
       52    0.000    0.000    0.000    0.000 {method 'keys' of 'dict' objects}
       54    0.000    0.000    0.000    0.000 {method 'items' of 'dict' objects}
        6    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/logging/__init__.py:1367(isEnabledFor)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:774(<lambda>)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:77(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3098(_from_objects)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/abc.py:128(__instancecheck__)
       51    0.000    0.000    0.000    0.000 {method 'iteritems' of 'dict' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1329(expunge_all)
       51    0.000    0.000    0.000    0.000 <string>:1(set)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/operators.py:1486(is_precedent)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:2914(<genexpr>)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:458(_prepare_impl)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:1048(post_exec)
       51    0.000    0.000    0.000    0.000 /Users/czumar/mlflow/mlflow/entities/experiment.py:19(<dictcomp>)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:2291(_polymorphic_properties)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/state.py:575(_expire)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:733(__missing__)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:318(_set_lazyload_from)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1745(_do_commit)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:596(begin)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1564(_process_anon)
       53    0.000    0.000    0.000    0.000 {method 'close' of 'sqlite3.Cursor' objects}
       51    0.000    0.000    0.000    0.000 {method 'issuperset' of 'frozenset' objects}
       12    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:330(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/declarative/api.py:815(_inspect_decl_meta)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/state.py:309(_pending_mutations)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:436(iterkeys)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:63(set)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:623(<genexpr>)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:173(setup_query)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:67(bakery)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:1908(self_group)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:752(_commit_impl)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3975(_adjust_for_single_inheritance)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1737(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3127(self_group)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/base.py:409(_inspect_mapped_class)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:55(_interpret_as_from)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:845(_reset)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4139(__new__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:926(_manage_size)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:989(_path_registry)
       58    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:731(mapper)
      171    0.000    0.000    0.000    0.000 {method 'pop' of 'dict' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:115(__setitem__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:1217(_clone)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:625(<genexpr>)
        5    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:2299(_iterate_polymorphic_properties)
        5    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4467(_literal_as_text)
       51    0.000    0.000    0.000    0.000 {method 'difference_update' of 'set' objects}
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4447(_expression_literal_as_text)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:858(_reset)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:410(__init__)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/base.py:601(__contains__)
       16    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3941(_get_table)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4159(_memoized_method_lower)
       11    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:1145(<genexpr>)
       52    0.000    0.000    0.000    0.000 {callable}
       14    0.000    0.000    0.000    0.000 {method 'update' of 'set' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3205(_from_objects)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4340(_select_iterables)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:1068(__getattr__)
       46    0.000    0.000    0.000    0.000 {method 'startswith' of 'str' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:134(per_mapper)
        6    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/logging/__init__.py:1353(getEffectiveLevel)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4559(_interpret_as_column_or_from)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/declarative/clsregistry.py:182(<lambda>)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:2311(visit_table)
        1    0.000    0.000    0.000    0.000 <string>:1(_adapt_all_clauses)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:2055(iterate_properties)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:264(traverse_using)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:1904(_from_objects)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:861(reset)
       69    0.000    0.000    0.000    0.000 {method 'add' of 'set' objects}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1570(bindparam_string)
        4    0.000    0.000    0.000    0.000 {method 'match' of '_sre.SRE_Pattern' objects}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1168(_get_operator_dispatch)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:253(_inspect_func_args)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:1045(pre_exec)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:713(_begin_impl)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:419(type)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:296(__getitem__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:2263(selectable)
        5    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4451(_literal_as)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:123(_join)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:879(__init__)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:332(remove)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:534(coerce_compared_value)
       52    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:145(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:504(do_commit)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:261(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:316(values)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3950(_from_objects)
       11    0.000    0.000    0.000    0.000 {method 'get' of 'dictproxy' objects}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/base.py:228(manager_of_class)
        1    0.000    0.000    0.000    0.000 <string>:1(_with_invoke_all_eagers)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:307(_bake_subquery_loaders)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:1705(_prop_set)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3462(_select_args)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1758(_add_to_result_map)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:368(__setitem__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4059(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/_weakrefset.py:26(__exit__)
      3/2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/declarative/clsregistry.py:159(_remove_item)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:310(__new__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/mlflow/mlflow/entities/lifecycle_stage.py:10(view_type_to_stages)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:580(get_connection)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/_weakrefset.py:70(__contains__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:473(_cached_bind_processor)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4414(_clause_element_as_expr)
        2    0.000    0.000    0.000    0.000 {method 'split' of 'unicode' objects}
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:214(__contains__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/log.py:59(_should_log_info)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:958(__getattr__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:2823(_resolve_value_to_type)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:326(_adapt_clause)
       16    0.000    0.000    0.000    0.000 {_weakref._remove_dead_weakref}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4520(_is_literal)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3976(_render_label_in_columns_clause)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/schema.py:1616(get_children)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:307(_has_column_expression)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/identity.py:17(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:501(do_rollback)
        6    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:400(__contains__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:140(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/log.py:56(_should_log_debug)
        1    0.000    0.000    0.000    0.000 listexpprof.py:6(profiled)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:396(__iter__)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/interfaces.py:507(_memoized_attr__default_path_loader_key)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3180(__init__)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/interfaces.py:501(_memoized_attr__wildcard_token)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3965(_key_label)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/annotation.py:85(_clone)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/identity.py:207(all_states)
       12    0.000    0.000    0.000    0.000 {method 'discard' of 'set' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/contextlib.py:82(helper)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1674(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:315(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:486(_no_limit_offset)
        3    0.000    0.000    0.000    0.000 {method 'remove' of 'set' objects}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:2120(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:294(result_processor)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:600(visit_bindparam)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:2132(_with_polymorphic_mappers)
        3    0.000    0.000    0.000    0.000 {method 'difference' of 'set' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:261(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:1051(module)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/_weakrefset.py:20(__enter__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/UserDict.py:43(clear)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/relationships.py:2127(_use_get)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:596(visit_bindparam)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/interfaces.py:744(setup_query)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/contextlib.py:12(__init__)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:751(_select_iterable)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:305(_is_transaction_boundary)
        3    0.000    0.000    0.000    0.000 {method 'popleft' of 'collections.deque' objects}
        1    0.000    0.000    0.000    0.000 {range}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/baked.py:47(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/attr.py:279(exec_once)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:715(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3106(get_children)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:143(selectable)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:322(_adapt_all_clauses)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/_weakrefset.py:16(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4332(_expand_cloned)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4676(<genexpr>)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/instrumentation.py:102(is_mapped)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/UserDict.py:34(__len__)
        5    0.000    0.000    0.000    0.000 {method 'append' of 'collections.deque' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1679(_parent)
        4    0.000    0.000    0.000    0.000 {iter}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:730(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:318(_iterate_self_and_parents)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:2267(get_select_precolumns)
        1    0.000    0.000    0.000    0.000 {time.time}
        4    0.000    0.000    0.000    0.000 {method 'lower' of 'unicode' objects}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:474(_no_statement_condition)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:346(_copy_internals)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:1933(_from_objects)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/base.py:34(_clone)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py:103(_supports_population)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3184(self_group)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:498(do_begin)
        2    0.000    0.000    0.000    0.000 {method 'pop' of 'list' objects}
        1    0.000    0.000    0.000    0.000 {method 'iterkeys' of 'dict' objects}
        1    0.000    0.000    0.000    0.000 {method 'rollback' of 'sqlite3.Connection' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:2123(_single_table_criterion)
        2    0.000    0.000    0.000    0.000 {method 'union' of 'set' objects}
        1    0.000    0.000    0.000    0.000 {thread.allocate_lock}
        1    0.000    0.000    0.000    0.000 {method 'clear' of 'dict' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:938(memo)
        1    0.000    0.000    0.000    0.000 {method 'clear' of 'collections.deque' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:2140(_with_polymorphic_selectable)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:359(get_children)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:346(_commit_removals)
        4    0.000    0.000    0.000    0.000 {method 'setdefault' of 'dict' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/attr.py:340(for_modify)
        1    0.000    0.000    0.000    0.000 {method 'commit' of 'sqlite3.Connection' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:1125(_with_invoke_all_eagers)
        6    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:370(connection)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:269(result_processor)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:168(__getitem__)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4192(apply_map)
        1    0.000    0.000    0.000    0.000 {method 'disable' of '_lsprof.Profiler' objects}
```

and here are the contents of `new_profile.txt`:
```
         3608 function calls (3515 primitive calls) in 0.007 seconds

   Ordered by: cumulative time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.000    0.000    0.007    0.007 /Users/czumar/mlflow/mlflow/tracking/client.py:96(list_experiments)
        1    0.000    0.000    0.007    0.007 /Users/czumar/mlflow/mlflow/store/sqlalchemy_store.py:284(list_experiments)
        1    0.000    0.000    0.006    0.006 /Users/czumar/mlflow/mlflow/store/sqlalchemy_store.py:262(_list_experiments)
        1    0.000    0.000    0.006    0.006 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3154(all)
        1    0.000    0.000    0.003    0.003 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3311(__iter__)
       52    0.000    0.000    0.003    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/loading.py:35(instances)
   102/51    0.000    0.000    0.002    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/loading.py:492(_instance)
        1    0.000    0.000    0.002    0.002 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3336(_execute_and_instances)
       51    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/instrumentation.py:337(new_instance)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3351(_get_bind_args)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3330(_connection_from_session)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1047(connection)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1126(_connection_for_bind)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:406(_connection_for_bind)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:2223(_contextual_connect)
       51    0.001    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/state.py:89(__init__)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:2259(_wrap_pool_connect)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:345(connect)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:748(_checkout)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:922(execute)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:285(_execute_on_connection)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1056(_execute_clauseelement)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:481(checkout)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/impl.py:236(_do_get)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:296(_create_connection)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:425(__init__)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:622(__connect)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/strategies.py:106(connect)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:452(connect)
        1    0.001    0.001    0.001    0.001 {_sqlite3.connect}
       51    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/loading.py:677(_populate_full)
       51    0.000    0.000    0.001    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:1981(load_collection_from_joined_new_row)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3820(_compile_context)
        1    0.000    0.000    0.001    0.001 <string>:1(<lambda>)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:399(compile)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:464(_compiler)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:500(__init__)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:274(__init__)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:349(process)
     22/1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:85(_compiler_dispatch)
        1    0.000    0.000    0.001    0.001 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1999(visit_select)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1163(_execute_context)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py:1879(init_state_collection)
        2    0.000    0.000    0.000    0.000 /Users/czumar/mlflow/mlflow/store/sqlalchemy_store.py:153(make_managed_session)
       51    0.000    0.000    0.000    0.000 /Users/czumar/mlflow/mlflow/store/dbmodels/models.py:68(to_mlflow_entity)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:1700(_create_eager_join)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/contextlib.py:21(__exit__)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py:1176(initialize)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:992(commit)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/util.py:974(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:490(commit)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py:1183(_initialize_collection)
        2    0.000    0.000    0.000    0.000 {method 'execute' of 'sqlite3.Cursor' objects}
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1761(_label_select_column)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:551(do_execute)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4151(row_processor)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/relationships.py:2139(_create_joins)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/relationships.py:3086(join_targets)
      2/1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/loading.py:337(_instance_processor)
      257    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py:268(__get__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:569(close)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4186(setup_context)
      2/1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/loading.py:287(_setup_entity_query)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:726(operate)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:780(visit_label)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:210(traverse)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:316(replacement_traverse)
     10/6    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/interfaces.py:546(setup)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:64(operate)
      3/1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:323(clone)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/instrumentation.py:315(initialize_collection)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:2186(_compose_select_body)
      3/2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/interfaces.py:556(create_row_processor)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:869(close)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:1195(fetchall)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:987(close)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:1433(setup_query)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:213(replace)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:831(_checkin)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:1941(create_row_processor)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:645(_finalize_fairy)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/deprecations.py:117(warned)
      4/1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/operators.py:515(in_)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py:210(operate)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:1092(get_result_proxy)
        5    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/util.py:783(replace)
      3/1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/operators.py:1252(in_op)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:714(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/properties.py:314(operate)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:740(_init_metadata)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/base.py:215(generate)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:507(checkin)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:215(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:1161(_fetchall_impl)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:366(_return_conn)
        1    0.000    0.000    0.000    0.000 {method 'fetchall' of 'sqlite3.Cursor' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/impl.py:233(_do_return_conn)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/util.py:764(_corresponding_column)
       11    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:852(__get__)
       10    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:834(visit_column)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:755(unique_list)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:529(close)
      5/2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:97(_compiler_dispatch)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:1512(options)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:615(__close)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/state.py:379(_cleanup)
        1    0.000    0.000    0.000    0.000 <string>:1(_options)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/default_comparator.py:158(_in_impl)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:257(_close_connection)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:503(corresponding_column)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:1527(_options)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1261(visit_binary)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:507(do_close)
       51    0.000    0.000    0.000    0.000 /Users/czumar/mlflow/mlflow/entities/experiment.py:13(__init__)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/annotation.py:102(__eq__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3102(_copy_internals)
        1    0.000    0.000    0.000    0.000 {method 'close' of 'sqlite3.Connection' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:2157(_setup_select_stack)
       13    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/interfaces.py:516(_get_context_loader)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:167(process_query)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:517(_process)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3945(_simple_statement)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1318(_generate_generic_binary)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/contextlib.py:15(__enter__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:3188(__call__)
      6/3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/operators.py:358(__eq__)
        6    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/base.py:613(contains_column)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:606(_bind_loader)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:883(begin)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:2340(visit_join)
        1    0.000    0.000    0.000    0.000 <string>:1(select)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:3641(_columns_plus_names)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/default_comparator.py:41(_boolean_compare)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/collections.py:618(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:2988(_get_display_froms)
        6    0.000    0.000    0.000    0.000 {operator.eq}
        1    0.000    0.000    0.000    0.000 <string>:1(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1539(query)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:220(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:141(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:645(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:170(_set_entities)
      130    0.000    0.000    0.000    0.000 {isinstance}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:384(_remove_snapshot)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:2687(__init__)
        5    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:302(__missing__)
       50    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/identity.py:213(_fast_discard)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:775(__init__)
       11    0.000    0.000    0.000    0.000 {method 'update' of 'dict' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:2959(_froms)
        5    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:205(__init__)
       13    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1542(_truncated_identifier)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:638(_init_compiled)
      157    0.000    0.000    0.000    0.000 {getattr}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1278(close)
       14    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/registry.py:53(_collection_gced)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3066(__init__)
       30    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:3529(quote)
       21    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/inspection.py:38(inspect)
        1    0.000    0.000    0.000    0.000 /Users/czumar/mlflow/mlflow/store/sqlalchemy_store.py:313(_get_eager_experiment_query_options)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1323(_close_impl)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:717(visit_grouping)
      102    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/identity.py:163(get)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:1067(joinedload)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:336(_take_snapshot)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:529(_from_keys)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:961(visit_clauselist)
        3    0.000    0.000    0.000    0.000 {method 'join' of 'str' objects}
        8    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:250(__getitem__)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:173(setup_query)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/identity.py:158(_add_unpresent)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py:1330(get_collection)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:69(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4384(_find_columns)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/base.py:38(_from_objects)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:514(do_ping)
       13    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/util.py:862(__getitem__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:970(<genexpr>)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:183(_generate_path)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4019(_bind_param)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:968(<genexpr>)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:334(_merge_cursor_description)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:295(__get__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/annotation.py:168(_shallow_annotate)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:276(traverse)
       61    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/attr.py:255(__call__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:911(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:1913(_create_eager_adapter)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:975(joinedload)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:1636(_generate_row_adapter)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:254(_annotate)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/annotation.py:78(_compiler_dispatch)
      102    0.000    0.000    0.000    0.000 {method 'issuperset' of 'frozenset' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1445(visit_bindparam)
        5    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4299(apply_map)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:202(_set_entity_selectables)
        1    0.000    0.000    0.000    0.000 <string>:1(set_relationship_strategy)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/base.py:42(_generative)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:119(_for_instance)
       51    0.000    0.000    0.000    0.000 <string>:1(set)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:1178(process_rows)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:1864(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/base.py:346(_is_mapped_class)
        6    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/logging/__init__.py:1152(debug)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4037(__new__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/state.py:328(_detach_states)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1711(commit)
        1    0.000    0.000    0.000    0.000 <string>:1(filter)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:802(to_list)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:63(set)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:222(iterate)
       12    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:94(__getattr__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/declarative/clsregistry.py:106(_remove_item)
       13    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:56(remove)
        8    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:440(dialect_impl)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:2662(_result_has_identity_key)
       13    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4332(_expand_cloned)
        8    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:261(__init__)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:116(_for_class)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1705(visit_alias)
        8    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/logging/__init__.py:1367(isEnabledFor)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:1051(get_result_processor)
       51    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/instrumentation.py:328(get_impl)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:458(_prepare_impl)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1329(expunge_all)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:717(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/state.py:575(_expire)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:3598(format_label)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:361(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1745(_do_commit)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:752(_commit_impl)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:733(__missing__)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:77(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:596(begin)
        9    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:296(__getitem__)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:724(_getter)
       12    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:330(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/declarative/api.py:815(_inspect_decl_meta)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:436(iterkeys)
       19    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:362(__getitem__)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:55(_interpret_as_from)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1564(_process_anon)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:306(set_relationship_strategy)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:737(_find_entity_prop_comparator)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/base.py:409(_inspect_mapped_class)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1529(_truncate_bindparam)
        8    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:681(self_group)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3474(_should_nest_selectable)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:1908(self_group)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:3646(name_for_col)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:361(_set_for_path)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3086(__bool__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3098(_from_objects)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4642(__init__)
        8    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/logging/__init__.py:1353(getEffectiveLevel)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:1157(_next)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1737(__init__)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:2417(flush)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:198(_clone)
       20    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/attr.py:267(__bool__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/abc.py:128(__instancecheck__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:845(_reset)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4071(setup_entity)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:483(_cached_result_processor)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:643(_getter)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:620(_bind_processors)
      110    0.000    0.000    0.000    0.000 {method 'pop' of 'dict' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:115(__setitem__)
       60    0.000    0.000    0.000    0.000 {built-in method __new__ of type object at 0x1029d3bb8}
        5    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/operators.py:1399(is_boolean)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:3941(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:1034(create_cursor)
       13    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/util.py:389(_include_fn)
      109    0.000    0.000    0.000    0.000 {id}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/declarative/clsregistry.py:182(<lambda>)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:1729(filter)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/operators.py:1486(is_precedent)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:2914(<genexpr>)
        5    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:2463(_is_clean)
       10    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:154(visitor_iterator)
       45    0.000    0.000    0.000    0.000 {method 'append' of 'list' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4184(__new__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4340(_select_iterables)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:246(entity_path)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/default_comparator.py:354(_check_literal)
       21    0.000    0.000    0.000    0.000 {method 'intersection' of 'set' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:713(_begin_impl)
       51    0.000    0.000    0.000    0.000 /Users/czumar/mlflow/mlflow/entities/experiment.py:19(<dictcomp>)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:68(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:157(_generate)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:623(<genexpr>)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/util.py:315(clause_is_present)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3205(_from_objects)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:504(do_commit)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:399(_bind_mapper)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:2311(visit_table)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/base.py:228(manager_of_class)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:1187(_hide_froms)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:264(traverse_using)
        9    0.000    0.000    0.000    0.000 {method 'copy' of 'dict' objects}
        8    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3941(_get_table)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/util.py:745(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:163(chain)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3462(_select_args)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1758(_add_to_result_map)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:1601(_gen_pooled_aliased_class)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:1193(_from_objects)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4467(_literal_as_text)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:123(_join)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4447(_expression_literal_as_text)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:1430(to_instance)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:534(coerce_compared_value)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/annotation.py:52(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:1904(_from_objects)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:1980(_create_collection_loader)
      3/2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/ext/declarative/clsregistry.py:159(_remove_item)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:310(__new__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:625(<genexpr>)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3127(self_group)
       36    0.000    0.000    0.000    0.000 {method 'get' of 'dict' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:949(cursor)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1168(_get_operator_dispatch)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/log.py:59(_should_log_info)
        5    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/identity.py:56(check_modified)
       13    0.000    0.000    0.000    0.000 {_weakref._remove_dead_weakref}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:869(_soft_close)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:518(_clone)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:301(_coerce_strat)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:332(remove)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:858(_reset)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:501(do_rollback)
       75    0.000    0.000    0.000    0.000 {method 'add' of 'set' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:3975(_adjust_for_single_inheritance)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:410(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:1527(_init_user_defined_eager_proc)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/base.py:324(_generate)
       20    0.000    0.000    0.000    0.000 {len}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1573(_autoflush)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/util.py:331(surface_selectables)
        6    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:71(get)
        7    0.000    0.000    0.000    0.000 {hash}
        8    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:751(_select_iterable)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:267(_assert_active)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:1416(get_bind)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/util.py:1224(_entity_corresponds_to)
       10    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/annotation.py:99(__hash__)
        8    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4559(_interpret_as_column_or_from)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:958(__getattr__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/sqltypes.py:2823(_resolve_value_to_type)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:232(_cloned_set)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/base.py:374(_is_aliased_class)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4139(__new__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/schema.py:1616(get_children)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:641(construct_params)
        3    0.000    0.000    0.000    0.000 {hasattr}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:732(_has_key)
        5    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/relationships.py:3231(__call__)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/util.py:740(<lambda>)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/log.py:56(_should_log_debug)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:1298(self_group)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/_weakrefset.py:26(__exit__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:947(should_autocommit)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3180(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/mlflow/mlflow/entities/lifecycle_stage.py:10(view_type_to_stages)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/annotation.py:38(__new__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4414(_clause_element_as_expr)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:3601(format_alias)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/loading.py:880(for_context)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/identity.py:207(all_states)
        2    0.000    0.000    0.000    0.000 {method 'split' of 'unicode' objects}
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/attributes.py:103(_supports_population)
        7    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:419(type)
        5    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/operators.py:1376(is_comparison)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4131(_get_entity_clauses)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategies.py:883(create_row_processor)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:326(_adapt_clause)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:305(_is_transaction_boundary)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4520(_is_literal)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/contextlib.py:82(helper)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1674(__init__)
        6    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:731(mapper)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:315(__init__)
       12    0.000    0.000    0.000    0.000 {method 'update' of 'set' objects}
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:143(selectable)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/base.py:326(_class_to_mapper)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/identity.py:17(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:533(_split_key)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:861(reset)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:379(_entity_zero)
        2    0.000    0.000    0.000    0.000 {method 'intersection' of 'frozenset' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4059(__init__)
        1    0.000    0.000    0.000    0.000 listexpprof.py:6(profiled)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1570(bindparam_string)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/base.py:261(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:521(embedded)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:2120(__init__)
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/interfaces.py:536(_get_strategy)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:580(get_connection)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/_weakrefset.py:70(__contains__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:854(flatten_iterator)
        2    0.000    0.000    0.000    0.000 {method 'cursor' of 'sqlite3.Connection' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/type_api.py:473(_cached_bind_processor)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:477(_generate_path)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/util.py:706(_memo)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/UserDict.py:43(clear)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/state.py:412(dict)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:1045(pre_exec)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/util.py:1244(_entity_corresponds_to_use_path_impl)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3184(self_group)
        1    0.000    0.000    0.000    0.000 {sorted}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/visitors.py:321(<setcomp>)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:271(__init__)
        4    0.000    0.000    0.000    0.000 {method 'discard' of 'set' objects}
        2    0.000    0.000    0.000    0.000 {method 'acquire' of 'thread.lock' objects}
        3    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4451(_literal_as)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:371(_query_entity_zero)
       12    0.000    0.000    0.000    0.000 {setattr}
        1    0.000    0.000    0.000    0.000 {range}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/langhelpers.py:938(memo)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/pool/base.py:715(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:78(__len__)
        3    0.000    0.000    0.000    0.000 {method 'items' of 'dict' objects}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:3106(get_children)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:486(_no_limit_offset)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:2263(selectable)
        1    0.000    0.000    0.000    0.000 {sqlalchemy.cutils._distill_params}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/_weakrefset.py:16(__init__)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:4676(<genexpr>)
       21    0.000    0.000    0.000    0.000 {method 'values' of 'dict' objects}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/instrumentation.py:102(is_mapped)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/UserDict.py:34(__len__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/attr.py:279(exec_once)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/weakref.py:346(_commit_removals)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:1361(_from_objects)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/relationships.py:3228(__init__)
        5    0.000    0.000    0.000    0.000 {method 'setdefault' of 'dict' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1327(_safe_close_cursor)
        2    0.000    0.000    0.000    0.000 {method 'get' of 'dictproxy' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:1342(_copy_internals)
        2    0.000    0.000    0.000    0.000 {iter}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/event/attr.py:340(for_modify)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:637(_has_key)
        2    0.000    0.000    0.000    0.000 {method 'close' of 'sqlite3.Cursor' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:2132(_with_polymorphic_mappers)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:396(__iter__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:2267(get_select_precolumns)
        1    0.000    0.000    0.000    0.000 {zip}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:1679(_parent)
        1    0.000    0.000    0.000    0.000 {time.time}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:352(__str__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:448(__init__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:474(_no_statement_condition)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/query.py:390(_mapper_entities)
        2    0.000    0.000    0.000    0.000 {next}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/_weakrefset.py:20(__enter__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:1002(_use_server_side_cursor)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:97(contains)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/session.py:318(_iterate_self_and_parents)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/interfaces.py:744(setup_query)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:498(do_begin)
        2    0.000    0.000    0.000    0.000 {method 'pop' of 'list' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/contextlib.py:12(__init__)
        1    0.000    0.000    0.000    0.000 {method 'iterkeys' of 'dict' objects}
        1    0.000    0.000    0.000    0.000 {method 'rollback' of 'sqlite3.Connection' objects}
        2    0.000    0.000    0.000    0.000 {method 'release' of 'thread.lock' objects}
        3    0.000    0.000    0.000    0.000 {method 'popleft' of 'collections.deque' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/path_registry.py:291(__bool__)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:373(self_group)
        1    0.000    0.000    0.000    0.000 {method 'union' of 'set' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:151(union)
        1    0.000    0.000    0.000    0.000 {method 'clear' of 'dict' objects}
        1    0.000    0.000    0.000    0.000 {method 'keys' of 'dict' objects}
        1    0.000    0.000    0.000    0.000 {method 'clear' of 'collections.deque' objects}
        4    0.000    0.000    0.000    0.000 {callable}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/result.py:864(_cursor_description)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:359(get_children)
        3    0.000    0.000    0.000    0.000 {method 'remove' of 'set' objects}
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/compiler.py:1755(get_render_as_alias_suffix)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/util/_collections.py:730(__init__)
        1    0.000    0.000    0.000    0.000 {method 'commit' of 'sqlite3.Connection' objects}
        1    0.000    0.000    0.000    0.000 {method 'difference' of 'set' objects}
        6    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/base.py:370(connection)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/strategy_options.py:474(_set_path_strategy)
        1    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/engine/default.py:1048(post_exec)
        5    0.000    0.000    0.000    0.000 {method 'append' of 'collections.deque' objects}
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/orm/mapper.py:2596(common_parent)
        4    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/elements.py:4192(apply_map)
        2    0.000    0.000    0.000    0.000 /Users/czumar/anaconda2/lib/python2.7/site-packages/sqlalchemy/sql/selectable.py:1933(_from_objects)
        1    0.000    0.000    0.000    0.000 {method 'disable' of '_lsprof.Profiler' objects}
```

The `ncalls` column indicates the number of times that a particular function listed in the `filename:lineno(function)` column was invoked during the call to `MlflowClient.list_experiments()`. In the old profile, we see that there are `53` invocations of `{method 'execute' of 'sqlite3.Cursor' objects}`: one invocation for each experiment (50 created experiments + the default experiment) + one invocation I'm not sure about. In the new profile, we only see `2` invocations of `{method 'execute' of 'sqlite3.Cursor' objects}`. Based on the SQLAlchemy documentation (https://docs.sqlalchemy.org/en/13/faq/performance.html#execution-slowness), it appears that each invocation of `sqlite3.Cursor.execute` corresponds to the execution of a query.

Similar results are reproducible on MySQL and Postgres.

Reply:
This is fantastic! Thank you.